morepath/app.py

```diff
@@ -67,28 +104,14 @@ class App(dectate.App):
     def __init__(self):
         pass
 
-    @reify
-    def lookup(self):
-        """Get the :class:`reg.Lookup` for this application.
-
-        :return: a :class:`reg.Lookup` instance.
-        """
-        # this in turn uses a cached lookup from the reg_registry
-        # the caching happens on the reg_registry and not here to
-        # ensure that each instance of App uses the same cache.
-        if not self.is_committed():
-            self.commit()
-        return self.config.reg_registry.caching_lookup
-
-    def set_implicit(self):
-        set_implicit(self.lookup)
-
     def request(self, environ):
```

Comment:
This is extra work each time a request is created, though it's minimal. It's just something I avoided in my branch by doing it in the reified ``publish`` instead, which only happens once after startup time.

What is the benefit to doing this in request? Since it depends on request I don't think it would stop an explicit commit from being needed in those cases where a test WSGI client isn't created at all but the Morepath API is addressed directly.